tests/test_working_copy_postgis.py

```diff
@@ -798,3 +798,25 @@ def test_checkout_custom_crs(
                 "    },",
                 "  ]",
             ]
+
+
+def test_init_without_postgis(cli_runner, data_archive, no_postgis_db, tmp_path):
+    """This tests asserts that when trying to import a spatial gpkg into a Postgres wc, which doesn't have Postgis enabled,
+    the correct error message is shown."""
+
+    # Get the URL for the PostgreSQL database without PostGIS
+    postgres_url = no_postgis_db.original_url + "/test_schema"
```

Comment:
This will stop working very quickly when Kart finds that this schema is already in use - Kart is too scared to start messing with somebody's existing data, it likes to create a new schema or use an empty schema.  
What you need is `new_postgis_db_schema` - which is a fixture in conftest.py - except you need an equivalent fixture called `new_no_postgis_db_schema`. This will take care of giving you a new empty schema each time you run the test. You can look at other tests which use the `new_postgis_db_schema` to get an idea of how it's used.

Here's how I would modify new_postgis_db_schema so that you also have a new_no_postgis_db_schema:
```
@pytest.fixture()
def new_postgis_db_schema(request, postgis_db):
    return _new_pg_schema_for_test(request, postgis_db)


@pytest.fixture()
def new_no_postgis_db_schema(request, no_postgis_db):
    return _new_pg_schema_for_test(request, no_postgis_db)


def _new_pg_schema_for_test(request, engine):
    @contextlib.contextmanager
    def ctx(create=False):
        sha = hashlib.sha1(request.node.nodeid.encode("utf8")).hexdigest()[:20]
        schema = f"kart_test_{sha}"
        with engine.connect() as conn:
            # Start by deleting in case it is left over from last test-run...
            conn.execute(f"""DROP SCHEMA IF EXISTS "{schema}" CASCADE;""")
            # Actually create only if create=True, otherwise the test will create it
            if create:
                conn.execute(f"""CREATE SCHEMA "{schema}";""")
        try:
            url = urlsplit(engine.original_url)
            url_path = url.path.rstrip("/") + "/" + schema
            new_schema_url = urlunsplit(
                [url.scheme, url.netloc, url_path, url.query, ""]
            )
            yield new_schema_url, schema
        finally:
            # Clean up - delete it again if it exists.
            with engine.connect() as conn:
                conn.execute(f"""DROP SCHEMA IF EXISTS "{schema}" CASCADE;""")

    return ctx
```